docs/readthedocs/source/doc/Nano/QuickStart/pytorch_nano.md

```diff
@@ -0,0 +1,174 @@
+
+    return train_loader
+```
+
+### **Step 2: Define the Model**
+
+You may define your model in the same way as the standard PyTorch models.
+
+```python
+from torch import nn
+
+from bigdl.nano.pytorch.vision.models import vision
+
+class ResNet18(nn.Module):
+    def __init__(self, num_classes, pretrained=True, include_top=False, freeze=True):
+        super().__init__()
+        backbone = vision.resnet18(pretrained=pretrained, include_top=include_top, freeze=freeze)
+        output_size = backbone.get_output_size()
+        head = nn.Linear(output_size, num_classes)
+        self.model = nn.Sequential(backbone, head)
+
+    def forward(self, x):
+        return self.model(x)
+```
+
+### **Step 3: Define Train Loop**
+
+Suppose the custom train loop is as follows:
+
+```python
+import os
+import torch
+
+data_path = os.environ.get("DATA_PATH", ".")
+batch_size = 256
+max_epochs = 10
+lr = 0.01
+
+model = ResNet18(10, pretrained=False, include_top=False, freeze=True)
+loss_func = nn.CrossEntropyLoss()
+optimizer = torch.optim.Adam(model.parameters(), lr=lr)
+train_loader = create_dataloader(data_path, batch_size)
+
+model.train()
+
+for _i in range(max_epochs):
+    total_loss, num = 0, 0
+    for X, y in train_loader:
+        optimizer.zero_grad()
+        loss = loss_func(model(X), y)
+        loss.backward()
+        optimizer.step()
+        
+        total_loss += loss.sum()
+        num += 1
+    print(f'avg_loss: {total_loss / num}')
+```
+
+The `TorchNano` (`bigdl.nano.pytorch.TorchNano`) class is what we use to accelerate raw pytorch code. By using it, we only need to make very few changes to accelerate custom training loop.
+
+We only need the following steps:
+
+- define a class `MyNano` derived from our `TorchNano`
+- copy all lines of code into the `train` method of `MyNano`
+- add one line to setup model, optimizer and dataloader
+- replace the `loss.backward()` with `self.backward(loss)`
+
+```python
+import os
+import torch
+
+from bigdl.nano.pytorch import TorchNano
+
+class MyNano(TorchNano):
+    def train(self):
+        # copy all lines of code into this method
+        data_path = os.environ.get("DATA_PATH", ".")
+        batch_size = 256
+        max_epochs = 10
+        lr = 0.01
+
+        model = ResNet18(10, pretrained=False, include_top=False, freeze=True)
+        loss_func = nn.CrossEntropyLoss()
+        optimizer = torch.optim.Adam(model.parameters(), lr=lr)
+        train_loader = create_dataloader(data_path, batch_size)
+
+        # add this line to setup model, optimizer and dataloaders
+        model, optimizer, train_loader = self.setup(model, optimizer, train_loader)
+
+        model.train()
+
+        for _i in range(max_epochs):
+            total_loss, num = 0, 0
+            for X, y in train_loader:
+                optimizer.zero_grad()
+                loss = loss_func(model(X), y)
+                self.backward(loss)     # modify this line
```

Comment:
`self.backward` is still needed?

Reply:
yeah, if user only use `spawn`, `subprocess` and `ray` backend, then `self.backward` is not needed, but considering we may add more strategies in the future, the new strategies may need it, so we still added it

Reply:
> yeah, if user only use `spawn`, `subprocess` and `ray` backend, then `self.backward` is not needed, but considering we may add more strategies in the future, the new strategies may need it, so we still added it

We need to figure out a way to automatically check and enforce it if we need to ask the users to call these methods.

Reply:
ok, I'll think of it